webapp/package.json

```diff
@@ -7,38 +7,58 @@
         "build": "webpack --mode=production --display-error-details --progress",
         "lint": "eslint --ignore-pattern node_modules --ignore-pattern dist --ext .js --ext .jsx . --quiet",
         "fix": "eslint --ignore-pattern node_modules --ignore-pattern dist --ext .js --ext .jsx . --quiet --fix",
-        "test": "echo 'no webapp tests implemented';"
+        "test": "jest --forceExit --detectOpenHandles --verbose",
+        "test:watch": "jest --watch",
+        "test-ci": "jest --forceExit --detectOpenHandles --maxWorkers=2"
     },
     "dependencies": {
         "@primer/octicons-react": "^9.6.0",
         "core-js": "3.1.4",
+        "jest": "26.4.2",
         "mattermost-redux": "github:mattermost/mattermost-redux#b56e7e3a6f7f5de281fd3541eefec9345a8bbc89",
         "prop-types": "15.7.2",
         "react": "16.8.6",
         "react-bootstrap": "0.32.4",
+        "react-dom": "18.2.0",
         "react-markdown": "^4.3.1",
         "react-redux": "5.1.1",
         "redux": "4.0.1"
     },
     "devDependencies": {
         "@babel/cli": "7.5.0",
-        "@babel/core": "7.5.0",
+        "@babel/core": "7.8.0",
         "@babel/plugin-proposal-class-properties": "7.5.0",
-        "@babel/plugin-syntax-dynamic-import": "7.2.0",
         "@babel/plugin-proposal-optional-chaining": "7.11.0",
+        "@babel/plugin-syntax-dynamic-import": "7.2.0",
         "@babel/polyfill": "7.4.4",
         "@babel/preset-env": "7.5.0",
         "@babel/preset-react": "7.0.0",
         "@babel/runtime": "7.5.1",
         "babel-eslint": "10.0.2",
+        "babel-jest": "29.5.0",
         "babel-loader": "8.0.6",
         "css-loader": "3.0.0",
+        "enzyme": "3.11.0",
+        "enzyme-adapter-react-16": "1.15.7",
+        "enzyme-to-json": "3.6.2",
         "eslint": "6.0.1",
         "eslint-import-resolver-webpack": "0.11.1",
         "eslint-plugin-import": "2.18.0",
         "eslint-plugin-react": "7.14.2",
+        "identity-obj-proxy": "3.0.0",
         "style-loader": "0.23.1",
         "webpack": "4.35.3",
         "webpack-cli": "3.3.5"
+    },
+    "jest": {
+        "transform": {
+            "^.+\\.[t|j]sx?$": "babel-jest"
+        },
+        "moduleNameMapper": {
+            "^.+\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "identity-obj-proxy",
+            "^.+\\.(css|less|scss)$": "identity-obj-proxy",
+            "^.*i18n.*\\.(json)$": "<rootDir>/tests/i18n_mock.json",
+            "^bundle-loader\\?lazy\\!(.*)$": "$1"
+        }
     }
```

Comment:
Keep the jest config in a separate `jest.config.(js|ts|....)` file instead of adding it in `package.json` file.